### ISSUE TITLE
TypeError: Cannot read property 'unsub' of undefined

### ISSUE THREAD

**Issue by pmilla1606** [rakannimer/react-firebase#5]
First of all: thank you for this library!

I'm getting the following error `TypeError: Cannot read property 'unsub' of undefined` when mounting/unmounting nested `FirebaseDatabaseNodes`.

The error occurs in this block:

```
if (path in this.state.dataTree) {
  this.state.dataTree[path].unsub && this.state.dataTree[path].unsub();
}
```

which I think is in: `~/react-firebase/modules/database/src/components/FirebaseDatabaseProvider.tsx` on line 116, but I'm not sure.

I think a quick fix would be to do something along the lines of:
``` 
this.state.dataTree[path] && this.state.dataTree[path].unsub && this.state.dataTree[path].unsub();
```
But that looks like compiled output so I'm not sure.

Have you encountered this? What are your recommendations on nesting database nodes?

**Comment by pmilla1606** [comment]
After some testing, it's happening even when the nodes are not nested.

**Comment by pmilla1606** [comment]
Welp. Problem resolved after updating to 0.2.31.

Sorry about the noise.
